base_attachment_object_storage/models/ir_attachment.py

```diff
@@ -89,12 +90,19 @@ def _save_in_db_anyway(self):
             return True
 
         # Binary fields
-        if self.res_field:
-            # Binary fields are stored with the name of the field in
-            # 'res_field'
-            # 'image' fields can be rather large and should usually
-            # not be requests in bulk in lists
-            if self.res_field and self.res_field in self._local_fields:
+        data_to_store = self.datas
+        bin_data = base64.b64decode(data_to_store) if data_to_store else ''
+        current_mimetype = guess_mimetype(bin_data)
+        mimetypes_settings = self.env['ir.config_parameter'].sudo().get_param(
+            'mimetypes.list.storedb')
+        mimetypes_for_db_store = mimetypes_settings \
+            and mimetypes_settings.split(',') or []
```

Comment:
Avoid ` ... and ... or ...`.
`IrConfigParameter.get_param()` accepts a `default` argument, so you should be able to do:

```
        mimetypes_settings = self.env['ir.config_parameter'].sudo().get_param(
            'mimetypes.list.storedb', default=''
        )
        mimetypes_for_db_store = mimetypes_settings.split(',')
```